From 241127a67f7a10a93985442dab3a604f26e462ff Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Tue, 22 Dec 2020 12:21:06 +0200
Subject: [PATCH] [SW-20154] efuse: make read register platform specific

Efuse register's name has been modified in Gaudi2,
since the backward compatible name for this register
which is "read" has been made a reserved. Hence
it has been changed to "read_ctrl". To make such
usage possible, EFUSE_READ register offset is being
made platform specific and will be updated in the
platform specific files

Change-Id: Ib25f1758ec68d4325c4f5973d4c17455a8105271
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/Makefile b/Makefile
index 10d2d98..4a0f650 100644
--- a/Makefile
+++ b/Makefile
@@ -41,6 +41,7 @@
 
 ARCBOOT_COMMON_SRCS_C := $(shell ls common/*.c $(HL_DEV)/$(HL_TARGET).c)
 ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV)/security.c)
+ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV)/common.c)
 
 ARCBOOT_RAM_SRCS_C := app/arcboot_ram.c
 ARCBOOT_RAM_SRCS_S := app/crt0_ram.S
diff --git a/common/efuse.c b/common/efuse.c
index c438b6c..0abc3ff 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -26,19 +26,22 @@
 int fuse_init(void)
 {
 	int i;
+	uint32_t read_reg;
 
 	if (arcboot_rom_global.efuse_read == 1)
 		return -1;
+
+	read_reg = get_efuse_read_reg();
 
 	/* TODO: ask suyash to test time of this function */
 	/* TODO: facto 10 of function time ~10mS*/
 	regmap_writel(EFUSE_READ_RANGE_ALL,
 		      mmPSOC_DFT_EFUSE_READ_ADDRESS);
 
-	regmap_writel(1, mmPSOC_EFUSE_READ);
+	regmap_writel(1, read_reg);
 
 	for (i = 0; i < FUSE_RETRIES; i++)
-		if (regmap_readl(mmPSOC_EFUSE_READ))
+		if (regmap_readl(read_reg))
 			udelay(FUSE_DELAY_US);
 		else
 			break;
diff --git a/common/efuse.h b/common/efuse.h
index 11f54a1..6417f6e 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -52,5 +52,6 @@
 int fuse_read(uint32_t word, uint32_t *val);
 uint8_t efuse_otp_get_fw_version_security(void);
 void efuse_get_hash(uint32_t *key_hash);
+uint32_t get_efuse_read_reg(void);
 
 #endif /* _EFUSE_H_ */
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
new file mode 100644
index 0000000..fa3e927
--- /dev/null
+++ b/goya2_arc/common.c
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2020 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+#include <unistd.h>
+#include <efuse.h>
+#include <psoc_efuse_regs.h>
+
+uint32_t get_efuse_read_reg(void)
+{
+	return mmPSOC_EFUSE_READ;
+}
